core/genesis.go

```diff
@@ -256,14 +260,20 @@ func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
 }
 
 // ApplyOvmStateToState applies the initial OVM state to a state object.
-func ApplyOvmStateToState(statedb *state.StateDB) {
+func ApplyOvmStateToState(statedb *state.StateDB, xDomainMessengerAddress common.Address) {
 	for _, account := range vm.OvmStateDump.Accounts {
 		statedb.SetCode(account.Address, common.FromHex(account.Code))
 		statedb.SetNonce(account.Address, account.Nonce)
 		for key, val := range account.Storage {
 			statedb.SetState(account.Address, key, common.HexToHash(val))
 		}
 	}
+	// Set the storage slot associated with the cross domain messenger
+	// to the cross domain messenger address.
+	AddressManager := vm.OvmStateDump.Accounts["Lib_AddressManager"]
+	slot := common.HexToHash("0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e")
+	value := common.BytesToHash(xDomainMessengerAddress.Bytes())
+	statedb.SetState(AddressManager.Address, slot, value)
```

Comment:
Does this address turn into a left padded 32 byte value?

Reply:
Yes, see implementation here: https://github.com/ethereum-optimism/go-ethereum/blob/a6d53b67db7324585e61f30d472a6baf13b307b0/common/types.go#L115

`len(h)` is 32, `	copy(h[HashLength-len(b):], b)` says: copy starting at index (32 - 20) until the end, the bytes that come before (32 - 20) will be set to 0 because its the zero value

Reply:
Ahhh nice